core/primitives/src/transaction.rs

```diff
@@ -205,22 +205,31 @@ pub struct SignedTransaction {
     pub signature: Signature,
     #[borsh_skip]
     hash: CryptoHash,
+    #[borsh_skip]
+    size: u64,
 }
 
 impl SignedTransaction {
     pub fn new(signature: Signature, transaction: Transaction) -> Self {
-        let mut signed_tx = Self { signature, transaction, hash: CryptoHash::default() };
+        let mut signed_tx =
+            Self { signature, transaction, hash: CryptoHash::default(), size: u64::default() };
         signed_tx.init();
         signed_tx
     }
 
     pub fn init(&mut self) {
-        self.hash = self.transaction.get_hash();
+        let hash_and_size = self.transaction.get_hash_and_size();
```

Comment:
```rust
let (hash, size) = self.transaction.get_hash_and_size();
```
It is uncommon in Rust to not decompose the tuple, especially if you are accessing all elements.

Reply:
Yes, makes sense.